DOC-1105 RPCN on GCP BYOVPC (#238)
redpanda-data/cloud-docs: modules/get-started/pages/cluster-types/byoc/gcp/vpc-byo-gcp.adoc

  --source-ranges="<gke-master-cidr-range>" \
  --allow="tcp:9443,tcp:8443,tcp:6443"
```
+

> Replace `<gke-master-cidr-range>` with a /28 CIDR. For example: 10.0.7.240/28.

Reviewer: ```suggestion
Replace `<gke-master-cidr-range>` with a /28 CIDR. For example: 172.16.0.32/28. See Creating a private cluster with no client access to the public endpoint[https://cloud.google.com/kubernetes-engine/docs/how-to/legacy/network-isolation#private_cp] which describes the master CIDR range which is set using `--master-ipv4-cidr```

Author: Updated.

```
====

* Redpanda Connect service accounts
+
.Show commands

> [%collapsible]

Author: Please check all new configuration for accuracy.

====
```bash

> # Used to check for the existence of a Redpanda Connect Service Account with permissions to create pipelines.

Author: Updated.

> # Account used to check for the existence of secrets, which are required to create Redpanda Connect pipelines.

Reviewer 2: we need to add permissions to read the secrets in the API as well

Author: Updated

> gcloud iam service-accounts create redpanda-connect \ 

Author: Updated.

  --display-name="Redpanda Connect API Service Account" 

cat << EOT > redpanda-connect-api.role 
{

>   "name": "redpanda_redpanda_connect_api_role",

Author: Updated.

  "title": "Redpanda Connect API Role",  
  "description": "Redpanda Connect API Role", 
  "includedPermissions": [

>     "secretmanager.versions.access"

Reviewer 2: ```suggestion
    "secretmanager.secrets.get"
```

Reviewer 2: Lets also add `resourcemanager.projects.get` and `secretmanager.versions.access`, so in total it will be
```
"resourcemanager.projects.get",
"secretmanager.secrets.get",
"secretmanager.versions.access"
```

Author: Updated.

  ],
}
EOT

gcloud iam roles create redpanda_redpanda_connect_api_role --project=<service-project-id> --file redpanda-connect-api.role

gcloud projects add-iam-policy-binding <service-project-id> \
  --member="serviceAccount:redpanda-connect-api@<service-project-id>.iam.gserviceaccount.com" \

>   --role='projects/<service-project-id>/roles/redpanda_redpanda_connect_api_role'

Author: Updated.

```
```bash

> # Used to create Redpanda Connect pipelines.

Reviewer 2: The Redpanda Connect needs to retrieve secrets, so it has permissions required to read secrets.

Author: Updated.

```
====

* Redpanda Connect service accounts
+
.Show commands
[%collapsible]
====
```bash
# Account used to check for and read secrets, which are required to create Redpanda Connect pipelines.

> gcloud iam service-accounts create redpanda-connect-api \ 

Author: This is in a code block so not an appropriate comment.

  --display-name="Redpanda Connect API Service Account" 

cat << EOT > redpanda-connect-api.role 
{
  "name": "redpanda_connect_api_role",
  "title": "Redpanda Connect API Role",  
  "description": "Redpanda Connect API Role", 
  "includedPermissions": [
    "resourcemanager.projects.get",
    "secretmanager.secrets.get",
    "secretmanager.versions.access"
  ],
}
EOT

> gcloud iam roles create redpanda_connect_api_role --project=<service-project-id> --file redpanda-connect-api.role

Author: This is in a code block so not an appropriate comment.

```
====

* Redpanda Connect service accounts
+
.Show command
[%collapsible]
====
```
gcloud iam service-accounts add-iam-policy-binding <redpanda_connect_api-gcp-sa-account-id>@<service-project-id>.iam.gserviceaccount.com \
    --role roles/iam.workloadIdentityUser \

>     --member "serviceAccount:<service-project-id>.svc.id.goog[redpanda-connect-api/connectors-<redpanda_connect_api-gcp-sa-account-id>]"

Reviewer 2: ```suggestion
    --member "serviceAccount:<service-project-id>.svc.id.goog[redpanda-connect/<redpanda_connect_api-gcp-sa-account-id>]"
```

Author: Updated.

```
```
gcloud iam service-accounts add-iam-policy-binding <redpanda_connect-gcp-sa-account-id>@<service-project-id>.iam.gserviceaccount.com \
    --role roles/iam.workloadIdentityUser \

>     --member "serviceAccount:<service-project-id>.svc.id.goog[redpanda-connect/connectors-<redpanda_connect-gcp-sa-account-id>]"

Reviewer 2: ```suggestion
    --member "serviceAccount:<service-project-id>.svc.id.goog[redpanda-connect/<redpanda_connect-gcp-sa-account-id>]"
```

Author: Updated.

```
====

* Redpanda Connect service accounts
+
.Show command
[%collapsible]
====
```

> gcloud iam service-accounts add-iam-policy-binding <redpanda_connect_api-gcp-sa-account-id>@<service-project-id>.iam.gserviceaccount.com \

Author: This is in a code block so not an appropriate comment.

    --role roles/iam.workloadIdentityUser \
    --member "serviceAccount:<service-project-id>.svc.id.goog[redpanda-connect/<redpanda_connect_api-gcp-sa-account-id>]"
```
```

> gcloud iam service-accounts add-iam-policy-binding <redpanda_connect-gcp-sa-account-id>@<service-project-id>.iam.gserviceaccount.com \

Author: This is in a code block so not an appropriate comment.
